src/irc/core/irc-servers.c

```diff
@@ -71,13 +71,20 @@ static int isnickflag_func(SERVER_REC *server, char flag)
 
 static int ischannel_func(SERVER_REC *server, const char *data)
 {
-	if (*data == '@') {
-		/* @#channel, @+#channel */
-		data++;
-		if (*data == '+' && ischannel(data[1]))
-			return 1;
-	}
-	return ischannel(*data);
+	IRC_SERVER_REC *irc_server = (IRC_SERVER_REC *) server;
+	char *chantypes;
+
+	chantypes = g_hash_table_lookup(irc_server->isupport, "chantypes");
+	if (chantypes == NULL)
+		chantypes = "#&!+"; /* normal, local, secure, modeless */
+
+	/* @#channel, @+#channel */
+	if (data[0] == '@' && data[1] == '+')
```

Comment:
You got review'd from a different ticket: https://github.com/irssi/irssi/issues/253#issuecomment-106934788

>the logic introduced at [this line] does not account for the fact that an arbitrary number of channel types can be prefixed to the "#channel", including duplicates